plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -399,8 +397,8 @@ def _get_next_page_of_agents(self, results_object: dict) -> (bool, dict, list):
         :param results_object: dict
         :return: tuple (boolean, dict (results), list (agents))
         """
-        self.logger.info("Getting next page of agents.")
         next_cursor = results_object.get("data").get("organization").get("assets").get("pageInfo").get("endCursor")
+        self.logger.info(f"Getting next page of agents using cursor {next_cursor}")
         payload = {
             "query": "query( $orgId:String! $nextCursor:String! ) { organization(id: $orgId) { assets( first: 10000, after: $nextCursor ) { pageInfo { hasNextPage endCursor } edges { node { id platform host { vendor version description hostNames { name } primaryAddress { ip mac } uniqueIdentity { source id } attributes { key value } } agent { agentSemanticVersion agentStatus quarantineState { currentState } } } } } } }",
```

Comment:
looking at some other actions can this query not be tailored to query the agent value? I think we know the agent type, so could do something like 

```
"query( $orgId:String! $nextCursor:String!  $agentValue:String!).....
"variables": {"orgId": self.org_key, "nextCursor": next_cursor, "agentValue": agent_input },
```

Just seems weird the query API can't take any input other than the org ID being requested!

Reply:
From what I've read and tried on the endpoint, it doesn't seem possible. The last recommended advice was to query for all agents then filter. Following up through the API docs now to see if there's any update.

Reply:
No this is not supported.

Reply:
Annoying we can try to improve the query but worth giving less logging a go to see if it helps as that seems to be the change between versions that are running successfully for the customer.